core/src/mender-client.c

```diff
@@ -1138,6 +1187,16 @@ mender_client_update_work_function(void) {
 
 #ifdef CONFIG_MENDER_FULL_PARSE_ARTIFACT
 #ifdef CONFIG_MENDER_PROVIDES_DEPENDS
+    /* Compare Artifact's depends with the stored provides */
+    if (MENDER_OK != mender_check_device_compatibility(mender_artifact_ctx)) {
+        /* Errors logged by function */
+        mender_client_publish_deployment_status(deployment->id, MENDER_DEPLOYMENT_STATUS_FAILURE);
+        if (mender_client_deployment_needs_set_pending_image) {
+            mender_flash_abort_deployment(mender_client_flash_handle);
+        }
+        goto END;
+    }
+
```

Comment:
OK for now because we are refactoring this part of the code, but as I asked for before -- please don't add more code into this function. We need to merge this with `mender_compare_device_types()` into a more comprehensive `check()` function.